src/main/java/com/google/sps/servlets/AggregationServlet.java

```diff
@@ -66,19 +67,52 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     // Some mock devices so we have data in the interim
     devices.add(new ChromeOSDevice("assetId", "location", "user", "deviceId", "serialNumber"));
     devices.add(new ChromeOSDevice("12345", "California", "Jane", "ae25f1-91ce6a", "SN54321"));
+    devices.add(new ChromeOSDevice("12345", "New Jersey", "Jane", "<deviceId>", "SN12345"));
 
     return devices;
   }
 
-  public Map<String, Integer> processData(List<ChromeOSDevice> devices, AnnotatedField field) {
-    Map<String, Integer> counts = new HashMap<>();
+  public static MultiKeyMap<String, Integer> processData(List<ChromeOSDevice> devices, Set<AnnotatedField> fields) {
+    MultiKeyMap<String, Integer> counts = new MultiKeyMap<>();
```

Comment:
Can we use `AnnotatedField` as key type directly instead of `String`? Then we don't have to convert String<=> AnnotatedField back and forth.

Also, when you construct `AggregationResponseEntry`, if `AnnotatedField` is the key, you don't have to pass in `Set<AnnotatedField> fields`.

The main concern is, the `Set<AnnotatedField>` was iterated in several places. I am not sure if the elements in Set will always be iterated in the same order. Iterating through a `List` could be more reassuring.

Reply:
+1 to make the key to AnnotatedField.

Reply:
AnnotatedField is just an enum containing the 3 different types of annotated fields (asset id, location, user).  It doesn't actually store any data as to what the values of those fields are.  Making the key a `String` means that the key can be something like `[Justin, Texas]` if the user is aggregating by user and location.  The `Set<AnnotatedField> fields` is passed in because the key doesn't contain any information as to what its components are (e.g. it can't know that 'Justin' is a user and 'Texas' is a location).
As far as the iteration order concern goes I haven't seen any indication that it's in a different order every time in my tests but I could use a `LinkedHashSet` instead which seems to preserve insertion order.